### ISSUE TITLE
Add support for EverDrive-64 X7

### ISSUE THREAD

**Issue by Polprzewodnikowy** [Polprzewodnikowy/N64FlashcartMenu#4]

**Comment by Polprzewodnikowy** [comment]
Yes, i think it is possible to support ED64 clones as libdragon already incorporates [libcart](https://github.com/devwizard64/libcart) which claims to support these carts. SD card access shouldn't be a problem but one big issue before adding support for anything other than SC64 is figuring out how other flashcarts actually load data. Sadly there's little to none documentation of many registers used by ED64. There's altra64 code but as you already mentioned it has issues with some of the ROMs.

To summarize, ED64 support depends only on figuring out how to load data into the flashcart and correctly set specific registers. Anything other than that is generic to all flashcarts and should be already supported.

**Comment by riggles1** [comment]
Ah that makes sense, I'd suspect the big hurdle with that then would be flashram games as only ED64 clone firmware knows how to grab that save type. 

Would be neat to see some support at least I suspect a lot of these are basically dead weighs in people's collections. Appreciate the reply! Will be testing.

**Comment by networkfusion** [comment]
I have done some work towards this here: https://github.com/networkfusion/N64FlashcartMenu/tree/ed64-vseries . I am unsure of others progress in the area, but IMHO it should enable a Proof of concept.

**Comment by networkfusion** [comment]
I have now added a draft PR. If further help is needed with the implementation, please let me know.

**Comment by SlaughterRidge** [comment]
I can confirm that PR44 does at the very least run roms on the EverDrive 2.0. I tested roms that require the expansion pack and ones that do not require it.

**Comment by networkfusion** [comment]
> EDIT: to add to this, saving does not appear to work, at least for the games I tested. The EverDrive 2.0/2.5 requires using the reset button to write save files to the SD.

Are you suggesting that the last played game is saving, but only when you have pressed reset? or it is still not working regardless of when you press reset?

**Comment by SlaughterRidge** [comment]
Sorry, to clarify, the EverDrive 2.0 and 2.5 require the user to press the reset button after the game itself saves. This is a requirement when using the official OS. 

Using this build, the save files get create when launching the ROM (I can see them in the /saves directory) but pressing the reset button doesn't appear to write the save to the file. Upon re-launching the game, the save file isn't present within the game itself - in my case, I was using the Ocarina of Time as a test.

**Comment by networkfusion** [comment]
Thanks for the clarification. The PR does take into account the need to save on reset. But, was only tested on an ED64-P and also there has been a few upstream changes that could have broken it (since I merged). Rather than anymore noise on this issue, it would be better to note comments against the PR, and if you are able, help to get it to a state where it is able to be merged, it would be welcome.

**Comment by networkfusion** [comment]
> Ocarina of Time

Also, since ROM's use multiple save types, it is possible that other ROM's are able to save correctly. Please test and show results for others that use `EEP4K`, `EEP16K`, `SRAM` , `FRAM`.

**Comment by lewilou22** [comment]
https://github.com/lewilou22/N64FlashcartMenu/releases/tag/V.01

V2 everdrive saving support
